### PR TITLE
[3.x][Filecache] Adding mounting and Unmounting for Filecache

### PR DESCRIPTION
**Pull request by himani2411**
### Description of changes

* Extending mounting and unmounting Logic for Fsx Filecache

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [ ] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb [hunk 1]
```diff
@@ -116,6 +116,11 @@ def initialize(node, resource, index)
         @mount_options = %w(defaults _netdev flock user_xattr noatime noauto x-systemd.automount)
         @device_name = "#{@dns_name}@tcp:/#{@mount_name}"
         @fstype = 'lustre'
+
+      elsif @type == 'FILECACHE'
+        @mount_options = %w(defaults _netdev flock user_xattr noatime noauto x-systemd.automount x-systemd.requires=network.service)
```
Comment (dreambeyondorange) [inline cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb:121]: just for my own knowledge, why do you need to add this? `x-systemd.requires=network.service`

File: cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb [hunk 2]
```diff
@@ -116,6 +116,11 @@ def initialize(node, resource, index)
...
+        @device_name = "#{@dns_name}@tcp:/#{@mount_name}"
+        @fstype = 'lustre'
```
Comment (dreambeyondorange) [inline cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb:123]: Is file cache always a lustre filesystem?

Comment (himani2411) [reply cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb:123]: Yes

File: cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb (hunk 1 above)
Comment (himani2411) [reply cookbooks/aws-parallelcluster-common/resources/lustre/partial/_mount_unmount.rb:121]: This is one of the mount options which is required for automounting the FileCache. This specific option is
the one makes sure that the auto mounter does not run until the network connectivity is online.
https://docs.aws.amazon.com/fsx/latest/FileCacheGuide/mount-fs-auto-mount-onreboot.html

**Review by hanwen-pcluste — APPROVED** [review]

**Pull request merged**